server/src/main/java/io/crate/planner/optimizer/tracer/OptimizerProgressTracker.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.planner.optimizer.tracer;
+
+import io.crate.planner.operators.LogicalPlan;
+import io.crate.planner.optimizer.Rule;
+import io.crate.planner.optimizer.costs.PlanStats;
+
+public interface OptimizerProgressTracker {
```

Comment:
```suggestion
public interface OptimizerTracer {
```

Not sure about the naming. The variables in this PR use `tracer`. The classes `Tracker`. I think we should settle on one and avoid using two terms for the same thing. Either name sounds good to me.

I'm also not sure if we need the `Progress` part. I think the information that it is tracing optimization is good enough. (But no strong opinion if you want to keep the `Progress`)

Same would apply to the interface implementations

Reply:
Good point. I don't know why I overlooked this inconsistency, but I fixed it as you suggested.